vignettes/example_models.Rmd

```diff
@@ -33,14 +33,65 @@ Below are a few examples of common statistical models implemented in greta.
 
 <hr>
 
+```{r child='examples/linear_uninformative.Rmd'}
```

Comment:
It's great to have this demo of the different priors you can use in greta. I think they don't quite fit in the 'common models' section though.

Could you put them in a new section, with a title like 'common Bayesian priors', perhaps just showing the options for priors, like:
```r
 and a little discussion abou

Reply:
I just remembered I wrote the following a while ago. Might be worth adding them at some point too:

```r
# this is smoothed-out approximation to the true spike-and-slab prior that
# replaces the discrete indicator variable with a continuous variable.
# 'spikiness' sets the amount of spike on this, values closer to 0 should be
# closer to the discrete version, but may take longer to converge. Values
# greater than 1 will break everything. Don't do that.
spike_slab <- function (spikiness = 0.1, slabiness = 10, dim = NULL) {
  stopifnot(spikiness < 1)
  slab <- normal(0, slabiness, dim = dim)
  spike <- gamma(spikiness, spikiness, dim = dim)
  spike * slab
}

horseshoe <- function (tau = 1, dim = NULL) {
  lambda <- cauchy(0, 1, truncation = c(0, Inf), dim = dim)
  sd <- tau ^ 2 * lambda ^ 2
  normal(zeros_like(sd), sd, dim = dim)
}

ponyshoe <- function (tau = 1, eta = 3, s = 1, dim = NULL) {
  tau2 <- tau ^ 2
  c2 <- inverse_gamma(alpha = eta / 2, beta = eta * s ^ 2)
  lambda <- cauchy(0, 1, truncation = c(0, Inf), dim = dim)
  lambda2 <- lambda ^ 2
  lambda_tilde <- (c2 * lambda2) / (c2 * tau2 * lambda2)
  sd <- tau2 * lambda_tilde ^ 2
  normal(zeros_like(sd), sd, dim = dim)
}
```